chalicelib/checks/wfr_checks.py

```diff
@@ -1,1448 +1,1456 @@
+    """
+    found = []
+    not_found = []
+    if isinstance(item_identifiers, str):
+        item_identifiers = [item_identifiers]
+    for item_identifier in item_identifiers:
+        try:
+            item = ff_utils.get_metadata(
+                item_identifier,
+                key=connection.ff_keys,
+                add_on="frame=raw&datastore=database",
+            )
+            found.append(item)
+        except Exception:
+            not_found.append(item_identifier)
+    return found, not_found
+
+
+def add_to_dict_as_list(dictionary, key, value):
+    """Add key, value pair to dictionary, with values for key stored in
+    list.
+    """
+    existing_item_value = dictionary.get(key)
+    if existing_item_value:
+        existing_item_value.append(value)
+    else:
+        dictionary[key] = [value]
+
+
+def get_and_add_field(items, field, list_to_append):
+    """Add field from items to given list."""
+    for item in items:
+        item_uuid = item.get(field)
+        if item_uuid is not None:
+            list_to_append.append(item_uuid)
+
+
+def get_and_add_uuids(items, list_to_append):
+    """Add UUIDs from items to given list."""
+    get_and_add_field(items, "uuid", list_to_append)
+
+
+def make_embed_request(ids, fields, connection):
+    """POST to /embed API to get desired fields for all given
+    identifiers.
+    """
+    result = []
+    if isinstance(ids, str):
+        ids = [ids]
+    if isinstance(fields, str):
+        fields = [fields]
+    id_chunks = chunk_ids(ids)
+    for id_chunk in id_chunks:
+        post_body = {"ids": ids, "fields": fields}
+        endpoint = connection.ff_server + "embed"
+        embed_response = ff_utils.authorized_request(
+            endpoint, verb="POST", auth=connection.ff_keys, data=json.dumps(post_body)
+        ).json()
+        result += embed_response
+    if len(result) == 1:
+        result = result[0]
+    return result
+
+
+def chunk_ids(ids):
+    """Split list into list of lists of maximum chunk size length.
+
+    Embed API currently accepts max 5 identifiers, so chunk size is 5.
+    """
+    result = []
+    chunk_size = 5
+    for idx in range(0, len(ids), chunk_size):
+        result.append(ids[idx: idx + chunk_size])
+    return result
+
+
+@check_function(file_type="File", start_date=None)
+def md5runCGAP_status(connection, file_type="", start_date=None, **kwargs):
+    """
+    Searches for files that are uploaded to s3, but not went through md5 run
+    Check assumptions:
+        - all files that have a status uploaded run through md5runCGAP
+        - all files status uploading/upload failed and NO s3 file are skipped
+
+    If the status is changed manually, it might fail to show up in this checkself
+
+    kwargs:
+        file_type -- limit search to a file type, i.e. FileFastq (default=File)
+        start_date -- limit search to files generated since date  YYYY-MM-DD
+        run_time -- assume runs beyond run_time are dead (default=24 hours)
```

Comment:
I don't see run_time referenced in the check, did you mean to add it to the decorator and/or the check logic?